solvebio/resource/object.py

```diff
@@ -0,0 +1,69 @@
+"""Solvebio Object API resource"""
+from ..help import open_help
+from ..client import client
+
+from .solveobject import convert_to_solve_object
+from .apiresource import CreateableAPIResource
+from .apiresource import ListableAPIResource
+from .apiresource import SearchableAPIResource
+from .apiresource import UpdateableAPIResource
+from .apiresource import DeletableAPIResource
+
+
+class Object(CreateableAPIResource,
+             ListableAPIResource,
+             DeletableAPIResource,
+             SearchableAPIResource,
+             UpdateableAPIResource):
+    """
+    An object is a resource in a Vault.  It has three possible types,
+    though more may be added later: folder, file, and SolveBio Dataset.
+    """
+    RESOURCE_VERSION = 2
+
+    LIST_FIELDS = (
+        ('id', 'ID'),
+        ('object_type', 'Object Type'),
+        ('path', 'Path'),
+        ('filename', 'Filename'),
+        ('description', 'Description'),
+    )
+
+    @classmethod
+    def retrieve_by_full_path(cls, full_path, **params):
```

Comment:
When creating a dataset, the user can pass a path, "/foo/bar/baz", and the code recursively checks if all directories exist, creating them along the way if they don't.  That check uses the this retrieve_by_path method, with an additional vault_id parameter passed in in **params.

Retrieve_by_full_path is used when a user attempts to retrieve a dataset using the "account:vault:path" format.  The user-provided value is passed straight through to the API, and the search is performed without any name-to-ID translations on the client.

Basically, one is designed to be used with additional filters, and one is not.